### ISSUE TITLE
Imaginary part of the wave function

### ISSUE THREAD

**Issue by RandomUsernameToSave** [quantum-visualizations/qmsolve#27]
In the time independent harmonic oscillator from example, the eigen states doesn't have any imaginary part contrary to what could be expected by the analytical resolution of the harmonic oscillator. (exp(i*m*phi))

Do you have any idea why, and how I could fix it ?

**Comment by rafael-fuente** [comment]
Hi,

The harmonic oscillator, as well as any other system subjected to a real value Hamiltonian, only has real-valued eigenstates up to an arbitrary complex constant with a module equal to 1.

Eigenstates of the [1D harmonic oscillator](https://en.wikipedia.org/wiki/Quantum_harmonic_oscillator) are:

![Screenshot 2023-11-17 173148](https://github.com/quantum-visualizations/qmsolve/assets/37953893/3c9cfacf-6245-4873-8978-227f011986b3)

Which they only have real part.

**Comment by RandomUsernameToSave** [comment]
Yes but the 3D harmonic oscillator has the spherical harmonics function : 
![image](https://github.com/quantum-visualizations/qmsolve/assets/101328862/4d51896c-e1a9-4b6a-a059-37e44384b40b)
![image](https://github.com/quantum-visualizations/qmsolve/assets/101328862/2c907ee1-88fe-4d49-b224-9e0bd1f4217f)

**Comment by rafael-fuente** [comment]
The complex spherical harmonic basis is not the unique basis of the 3D harmonic oscillator due to energy degeneracy. This solver will normally converge separately to the real  $\frac{Y^{m}_l + Y^{-m}_l}{2}$ and imaginary part   $\frac{Y^{m}_l - Y^{-m}_l}{2i}$ of the wavefunction you cite, which form a basis with the same span as the m = +-1 spherical harmonics $Y^{m}_l$ and $Y^{-m}_l$    



For the hydrogen atom, it will happen exactly the same: instead of converging to the complex spherical harmonics, it will converge to the basis it's usually used in chemistry books to represent p orbitals.   However, if you turn on a magnetic field, as illustrated in the [Zeeman effect example](https://github.com/quantum-visualizations/qmsolve/blob/main/examples/eigenstate%20solver%20examples/3D_hydrogen_atom_magnetic_field.py), the degeneracy breaks and the eigenstates will converge to wavefunctions with the angular part approaching the complex spherical harmonics.
